lightpath/ui/gui.py

```diff
@@ -274,6 +309,9 @@ def update_path(self, *args,  **kwargs):
                 # Update widget display
                 for widget in row:
                     widget.update_light(_in, _out)
+                    # Reconsider blocking device state
+                    if device is block:
+                        widget.update_state()
```

Comment:
Is this an optimization of some sort? Or is it a bugfix?

Reply:
Ah this was actually left over from when I colored downstream half-removed devices as half-removed, not blocking.  However this does remind me of a bug I'll fix shortly:

If you insert a bunch of attenuating devices, the last device gets colored red as expected.  But if you remove an upstream device and the final blocking device changes, the previous blocking device needs to change.